From 0ab435dc6accc9758333e4919f6484f74ea67c9d Mon Sep 17 00:00:00 2001
From: Luo Jie <quic_luoj@quicinc.com>
Date: Thu, 22 May 2025 11:21:12 +0800
Subject: [PATCH 504/512] arm64: dts: ipq9574: Update the IPQ9574 RDP433 DTS

Update the PCS & PPE dts nodes to align with the updated drivers.

Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
---
 arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts | 148 ++++++++----
 arch/arm64/boot/dts/qcom/ipq9574.dtsi       | 249 ++++++++++++--------
 2 files changed, 249 insertions(+), 148 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts b/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts
index fd5fe48ba917..9a04a3f653c7 100644
--- a/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts
+++ b/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts
@@ -69,6 +69,7 @@
 
 		phy0: ethernet-phy@10 {
 			reg = <0x10>;
+			clocks = <&cmn_pll ETH_25MHZ_CLK>;
 		};
 
 		phy1: ethernet-phy@11 {
@@ -86,21 +87,80 @@
 
 	phy4: ethernet-phy@8 {
 		compatible ="ethernet-phy-ieee802.3-c45";
+		clocks = <&cmn_pll ETH1_50MHZ_CLK>;
 		reg = <8>;
-
 		nvmem-cells = <&aqr_fw>;
 		nvmem-cell-names = "firmware";
 	};
 
 	phy5: ethernet-phy@0 {
 		compatible ="ethernet-phy-ieee802.3-c45";
+		clocks = <&cmn_pll ETH2_50MHZ_CLK>;
 		reg = <0>;
-
 		nvmem-cells = <&aqr_fw>;
 		nvmem-cell-names = "firmware";
 	};
 };
 
+&pcs0 {
+        status = "okay";
+};
+
+&pcs0_ch0 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT1_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT1_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs0_ch1 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT2_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT2_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs0_ch2 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT3_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT3_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs0_ch3 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT4_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT4_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs1 {
+        status = "okay";
+};
+
+&pcs1_ch0 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT5_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT5_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs2{
+        status = "okay";
+};
+
+&pcs2_ch0 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT6_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT6_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
 &tlmm {
 
 	pcie1_default: pcie1-default-state {
@@ -219,19 +279,19 @@
 			managed = "in-band-status";
 			label = "lan1";
 			phy-handle = <&phy0>;
-			pcs-handle = <&pcsuniphy0_ch0>;
+			pcs-handle = <&pcs0_ch0>;
 			clocks = <&nsscc NSS_CC_PORT1_MAC_CLK>,
 				 <&nsscc NSS_CC_PORT1_RX_CLK>,
 				 <&nsscc NSS_CC_PORT1_TX_CLK>;
-			clock-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			clock-names = "mac",
+				      "rx",
+				      "tx";
 			resets = <&nsscc PORT1_MAC_ARES>,
 				 <&nsscc PORT1_RX_ARES>,
 				 <&nsscc PORT1_TX_ARES>;
-			reset-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			reset-names = "mac",
+				      "rx",
+				      "tx";
 		};
 
 		port@2 {
@@ -240,19 +300,19 @@
 			managed = "in-band-status";
 			label = "lan2";
 			phy-handle = <&phy1>;
-			pcs-handle = <&pcsuniphy0_ch1>;
+			pcs-handle = <&pcs0_ch1>;
 			clocks = <&nsscc NSS_CC_PORT2_MAC_CLK>,
 				 <&nsscc NSS_CC_PORT2_RX_CLK>,
 				 <&nsscc NSS_CC_PORT2_TX_CLK>;
-			clock-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			clock-names = "mac",
+				      "rx",
+				      "tx";
 			resets = <&nsscc PORT2_MAC_ARES>,
 				 <&nsscc PORT2_RX_ARES>,
 				 <&nsscc PORT2_TX_ARES>;
-			reset-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			reset-names = "mac",
+				      "rx",
+				      "tx";
 		};
 
 		port@3 {
@@ -261,19 +321,19 @@
 			managed = "in-band-status";
 			label = "lan3";
 			phy-handle = <&phy2>;
-			pcs-handle = <&pcsuniphy0_ch2>;
+			pcs-handle = <&pcs0_ch2>;
 			clocks = <&nsscc NSS_CC_PORT3_MAC_CLK>,
 				 <&nsscc NSS_CC_PORT3_RX_CLK>,
 				 <&nsscc NSS_CC_PORT3_TX_CLK>;
-			clock-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			clock-names = "mac",
+				      "rx",
+				      "tx";
 			resets = <&nsscc PORT3_MAC_ARES>,
 				 <&nsscc PORT3_RX_ARES>,
 				 <&nsscc PORT3_TX_ARES>;
-			reset-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			reset-names = "mac",
+				      "rx",
+				      "tx";
 		};
 
 		port@4 {
@@ -282,19 +342,19 @@
 			managed = "in-band-status";
 			label = "lan4";
 			phy-handle = <&phy3>;
-			pcs-handle = <&pcsuniphy0_ch3>;
+			pcs-handle = <&pcs0_ch3>;
 			clocks = <&nsscc NSS_CC_PORT4_MAC_CLK>,
 				 <&nsscc NSS_CC_PORT4_RX_CLK>,
 				 <&nsscc NSS_CC_PORT4_TX_CLK>;
-			clock-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			clock-names = "mac",
+				      "rx",
+				      "tx";
 			resets = <&nsscc PORT4_MAC_ARES>,
 				 <&nsscc PORT4_RX_ARES>,
 				 <&nsscc PORT4_TX_ARES>;
-			reset-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			reset-names = "mac",
+				      "rx",
+				      "tx";
 		};
 
 		port@5 {
@@ -303,19 +363,19 @@
 			managed = "in-band-status";
 			label = "lan5";
 			phy-handle = <&phy4>;
-			pcs-handle = <&pcsuniphy1_ch0>;
+			pcs-handle = <&pcs1_ch0>;
 			clocks = <&nsscc NSS_CC_PORT5_MAC_CLK>,
 				 <&nsscc NSS_CC_PORT5_RX_CLK>,
 				 <&nsscc NSS_CC_PORT5_TX_CLK>;
-			clock-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			clock-names = "mac",
+				      "rx",
+				      "tx";
 			resets = <&nsscc PORT5_MAC_ARES>,
 				 <&nsscc PORT5_RX_ARES>,
 				 <&nsscc PORT5_TX_ARES>;
-			reset-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			reset-names = "mac",
+				      "rx",
+				      "tx";
 		};
 
 		port@6 {
@@ -324,19 +384,19 @@
 			managed = "in-band-status";
 			label = "wan";
 			phy-handle = <&phy5>;
-			pcs-handle = <&pcsuniphy2_ch0>;
+			pcs-handle = <&pcs2_ch0>;
 			clocks = <&nsscc NSS_CC_PORT6_MAC_CLK>,
 				 <&nsscc NSS_CC_PORT6_RX_CLK>,
 				 <&nsscc NSS_CC_PORT6_TX_CLK>;
-			clock-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			clock-names = "mac",
+				      "rx",
+				      "tx";
 			resets = <&nsscc PORT6_MAC_ARES>,
 				 <&nsscc PORT6_RX_ARES>,
 				 <&nsscc PORT6_TX_ARES>;
-			reset-names = "port_mac",
-				      "port_rx",
-				      "port_tx";
+			reset-names = "mac",
+				      "rx",
+				      "tx";
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/qcom/ipq9574.dtsi b/arch/arm64/boot/dts/qcom/ipq9574.dtsi
index bf30569cc30a..ddcb303d2156 100644
--- a/arch/arm64/boot/dts/qcom/ipq9574.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq9574.dtsi
@@ -1249,12 +1249,12 @@
 				 <&cmn_pll NSS_1200MHZ_CLK>,
 				 <&cmn_pll PPE_353MHZ_CLK>,
 				 <&gcc GPLL0_OUT_AUX>,
-				 <&pcs_uniphy0 0>,
-				 <&pcs_uniphy0 1>,
-				 <&pcs_uniphy1 0>,
-				 <&pcs_uniphy1 1>,
-				 <&pcs_uniphy2 0>,
-				 <&pcs_uniphy2 1>,
+				 <&pcs0 0>,
+				 <&pcs0 1>,
+				 <&pcs1 0>,
+				 <&pcs1 1>,
+				 <&pcs2 0>,
+				 <&pcs2 1>,
 				 <&gcc GCC_NSSCC_CLK>,
 				 <&gcc GCC_NSSNOC_NSSCC_CLK>,
 				 <&gcc GCC_NSSNOC_SNOC_CLK>,
@@ -1278,20 +1278,17 @@
 			#interconnect-cells = <1>;
 		};
 
-		qcom_ppe: ethernet@3a000000 {
+		qcom_ppe: ethernet-switch@3a000000 {
 			compatible = "qcom,ipq9574-ppe";
 			reg = <0x3a000000 0xbef800>;
-			ranges;
-			#address-cells = <1>;
-			#size-cells = <1>;
 			clocks = <&nsscc NSS_CC_PPE_SWITCH_CLK>,
 				 <&nsscc NSS_CC_PPE_SWITCH_CFG_CLK>,
 				 <&nsscc NSS_CC_PPE_SWITCH_IPE_CLK>,
 				 <&nsscc NSS_CC_PPE_SWITCH_BTQ_CLK>;
 			clock-names = "ppe",
-				      "ppe_cfg",
-				      "ppe_ipe",
-				      "ppe_btq";
+				      "apb",
+				      "ipe",
+				      "btq";
 			resets = <&nsscc PPE_FULL_RESET>;
 			interconnects = <&nsscc MASTER_NSSNOC_PPE
 					 &nsscc SLAVE_NSSNOC_PPE>,
@@ -1315,15 +1312,21 @@
 					     "memnoc_nssnoc",
 					     "memnoc_nssnoc_1";
 
-			edma {
-				compatible = "qcom,ipq9574-edma";
+			ethernet-dma {
 				clocks = <&nsscc NSS_CC_PPE_EDMA_CLK>,
 					 <&nsscc NSS_CC_PPE_EDMA_CFG_CLK>;
-				clock-names = "edma",
-					      "edma-cfg";
+				clock-names = "sys",
+					      "apb";
 				resets = <&nsscc EDMA_HW_RESET>;
-				reset-names = "edma_rst";
-				interrupts = <GIC_SPI 371 IRQ_TYPE_LEVEL_HIGH>,
+				interrupts = <GIC_SPI 363 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 364 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 365 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 366 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 367 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 368 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 369 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 370 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 371 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 372 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 373 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>,
@@ -1347,95 +1350,147 @@
 					     <GIC_SPI 502 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 501 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 500 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 355 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 356 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 357 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 358 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 359 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 360 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 361 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 362 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 331 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 332 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 333 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 335 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 336 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 337 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 339 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 340 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 341 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 343 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 344 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 345 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 346 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 347 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 348 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 349 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 350 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 351 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 352 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 353 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 354 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 499 IRQ_TYPE_LEVEL_HIGH>;
-				interrupt-names = "edma_txcmpl_8",
-						  "edma_txcmpl_9",
-						  "edma_txcmpl_10",
-						  "edma_txcmpl_11",
-						  "edma_txcmpl_12",
-						  "edma_txcmpl_13",
-						  "edma_txcmpl_14",
-						  "edma_txcmpl_15",
-						  "edma_txcmpl_16",
-						  "edma_txcmpl_17",
-						  "edma_txcmpl_18",
-						  "edma_txcmpl_19",
-						  "edma_txcmpl_20",
-						  "edma_txcmpl_21",
-						  "edma_txcmpl_22",
-						  "edma_txcmpl_23",
-						  "edma_txcmpl_24",
-						  "edma_txcmpl_25",
-						  "edma_txcmpl_26",
-						  "edma_txcmpl_27",
-						  "edma_txcmpl_28",
-						  "edma_txcmpl_29",
-						  "edma_txcmpl_30",
-						  "edma_txcmpl_31",
-						  "edma_rxdesc_20",
-						  "edma_rxdesc_21",
-						  "edma_rxdesc_22",
-						  "edma_rxdesc_23",
-						  "edma_misc";
+				interrupt-names = "txcmpl_0",
+						  "txcmpl_1",
+						  "txcmpl_2",
+						  "txcmpl_3",
+						  "txcmpl_4",
+						  "txcmpl_5",
+						  "txcmpl_6",
+						  "txcmpl_7",
+						  "txcmpl_8",
+						  "txcmpl_9",
+						  "txcmpl_10",
+						  "txcmpl_11",
+						  "txcmpl_12",
+						  "txcmpl_13",
+						  "txcmpl_14",
+						  "txcmpl_15",
+						  "txcmpl_16",
+						  "txcmpl_17",
+						  "txcmpl_18",
+						  "txcmpl_19",
+						  "txcmpl_20",
+						  "txcmpl_21",
+						  "txcmpl_22",
+						  "txcmpl_23",
+						  "txcmpl_24",
+						  "txcmpl_25",
+						  "txcmpl_26",
+						  "txcmpl_27",
+						  "txcmpl_28",
+						  "txcmpl_29",
+						  "txcmpl_30",
+						  "txcmpl_31",
+						  "rxfill_0",
+						  "rxfill_1",
+						  "rxfill_2",
+						  "rxfill_3",
+						  "rxfill_4",
+						  "rxfill_5",
+						  "rxfill_6",
+						  "rxfill_7",
+						  "rxdesc_0",
+						  "rxdesc_1",
+						  "rxdesc_2",
+						  "rxdesc_3",
+						  "rxdesc_4",
+						  "rxdesc_5",
+						  "rxdesc_6",
+						  "rxdesc_7",
+						  "rxdesc_8",
+						  "rxdesc_9",
+						  "rxdesc_10",
+						  "rxdesc_11",
+						  "rxdesc_12",
+						  "rxdesc_13",
+						  "rxdesc_14",
+						  "rxdesc_15",
+						  "rxdesc_16",
+						  "rxdesc_17",
+						  "rxdesc_18",
+						  "rxdesc_19",
+						  "rxdesc_20",
+						  "rxdesc_21",
+						  "rxdesc_22",
+						  "rxdesc_23",
+						  "misc";
 			};
 		};
 
-		pcs_uniphy0: ethernet-pcs@7a00000 {
-			compatible = "qcom,ipq9574-pcs";
-			reg = <0x7a00000 0x10000>;
+		pcs0: ethernet-pcs@7a00000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
+			compatible = "qcom,ipq9574-pcs";
+			reg = <0x7a00000 0x10000>;
 			clocks = <&gcc GCC_UNIPHY0_SYS_CLK>,
 				 <&gcc GCC_UNIPHY0_AHB_CLK>;
 			clock-names = "sys",
 				      "ahb";
-			resets = <&gcc GCC_UNIPHY0_SYS_RESET>,
-				 <&gcc GCC_UNIPHY0_AHB_RESET>,
-				 <&gcc GCC_UNIPHY0_XPCS_RESET>;
-			reset-names = "sys",
-				      "ahb",
-				      "xpcs";
-
+			resets = <&gcc GCC_UNIPHY0_XPCS_RESET>;
 			#clock-cells = <1>;
+			status = "disabled";
 
-			pcsuniphy0_ch0: pcs-mii@0 {
+			pcs0_ch0: pcs-mii@0 {
 				reg = <0>;
-				clocks = <&nsscc NSS_CC_UNIPHY_PORT1_RX_CLK>,
-					 <&nsscc NSS_CC_UNIPHY_PORT1_TX_CLK>;
-				clock-names = "rx",
-					      "tx";
+				status = "disabled";
 			};
 
-			pcsuniphy0_ch1: pcs-mii@1 {
+			pcs0_ch1: pcs-mii@1 {
 				reg = <1>;
-				clocks = <&nsscc NSS_CC_UNIPHY_PORT2_RX_CLK>,
-					 <&nsscc NSS_CC_UNIPHY_PORT2_TX_CLK>;
-				clock-names = "rx",
-					      "tx";
+				status = "disabled";
 			};
 
-			pcsuniphy0_ch2: pcs-mii@2 {
+			pcs0_ch2: pcs-mii@2 {
 				reg = <2>;
-				clocks = <&nsscc NSS_CC_UNIPHY_PORT3_RX_CLK>,
-					 <&nsscc NSS_CC_UNIPHY_PORT3_TX_CLK>;
-				clock-names = "rx",
-					      "tx";
+				status = "disabled";
 			};
 
-			pcsuniphy0_ch3: pcs-mii@3 {
+			pcs0_ch3: pcs-mii@3 {
 				reg = <3>;
-				clocks = <&nsscc NSS_CC_UNIPHY_PORT4_RX_CLK>,
-					 <&nsscc NSS_CC_UNIPHY_PORT4_TX_CLK>;
-				clock-names = "rx",
-					      "tx";
+				status = "disabled";
+			};
+
+			pcs0_ch4: pcs-mii@4 {
+				reg = <4>;
+				status = "disabled";
 			};
 		};
 
-		pcs_uniphy1: ethernet-pcs@7a10000 {
+		pcs1: ethernet-pcs@7a10000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			compatible = "qcom,ipq9574-pcs";
@@ -1444,46 +1499,32 @@
 				 <&gcc GCC_UNIPHY1_AHB_CLK>;
 			clock-names = "sys",
 				      "ahb";
-			resets = <&gcc GCC_UNIPHY1_SYS_RESET>,
-				 <&gcc GCC_UNIPHY1_AHB_RESET>,
-				 <&gcc GCC_UNIPHY1_XPCS_RESET>;
-			reset-names = "sys",
-				      "ahb",
-				      "xpcs";
 			#clock-cells = <1>;
+			resets = <&gcc GCC_UNIPHY1_XPCS_RESET>;
+			status = "disabled";
 
-			pcsuniphy1_ch0: pcs-mii@0 {
+			pcs1_ch0: pcs-mii@0 {
 				reg = <0>;
-				clocks = <&nsscc NSS_CC_UNIPHY_PORT5_RX_CLK>,
-					 <&nsscc NSS_CC_UNIPHY_PORT5_TX_CLK>;
-				clock-names = "rx",
-					      "tx";
+				status = "disabled";
 			};
 		};
 
-		pcs_uniphy2: ethernet-pcs@7a20000 {
-			compatible = "qcom,ipq9574-pcs";
-			reg = <0x7a20000 0x10000>;
+		pcs2: ethernet-pcs@7a20000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
+			compatible = "qcom,ipq9574-pcs";
+			reg = <0x7a20000 0x10000>;
 			clocks = <&gcc GCC_UNIPHY2_SYS_CLK>,
 				 <&gcc GCC_UNIPHY2_AHB_CLK>;
 			clock-names = "sys",
 				      "ahb";
-			resets = <&gcc GCC_UNIPHY2_SYS_RESET>,
-				 <&gcc GCC_UNIPHY2_AHB_RESET>,
-				 <&gcc GCC_UNIPHY2_XPCS_RESET>;
-			reset-names = "sys",
-				      "ahb",
-				      "xpcs";
+			resets = <&gcc GCC_UNIPHY2_XPCS_RESET>;
 			#clock-cells = <1>;
+			status = "disabled";
 
-			pcsuniphy2_ch0: pcs-mii@0 {
+			pcs2_ch0: pcs-mii@0 {
 				reg = <0>;
-				clocks = <&nsscc NSS_CC_UNIPHY_PORT6_RX_CLK>,
-					 <&nsscc NSS_CC_UNIPHY_PORT6_TX_CLK>;
-				clock-names = "rx",
-					      "tx";
+				status = "disabled";
 			};
 		};
 	};
-- 
2.34.1

